pkg/server/server.go

```diff
@@ -309,3 +324,25 @@ func mapConditions(conditions []metav1.Condition) []*pb.Condition {
 
 	return out
 }
+
+var BadErrProvider = errors.New("wrong provider name")
+
+// Authenticate generates and returns a jwt token using git provider name and git provider token
+func (s *applicationServer) Authenticate(_ context.Context, msg *pb.AuthenticateRequest) (*pb.AuthenticateResponse, error) {
+
+	if string(gitproviders.GitProviderGitHub) != msg.ProviderName &&
+		string(gitproviders.GitProviderGitLab) != msg.ProviderName {
+		return nil, status.Errorf(codes.InvalidArgument, "%s expected github or gitlab, got %s", BadErrProvider.Error(), msg.ProviderName)
```

Comment:
You don't need to call `.Error()` on the error here, because you're using `%s` https://pkg.go.dev/fmt

> 3. If the %v verb is used with the # flag (%#v) and the operand implements the GoStringer interface, that will be invoked.
> If the format (which is implicitly %v for Println etc.) is valid for a string (%s %q %v %x %X), the following two rules apply:
> 4. If an operand implements the error interface, the Error method will be invoked to convert the object to a string, which will then be formatted as required by the verb (if any).

So, because `BadErrProvider` implements `.Error()` it will be converted to a string automatically.